pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ComponentDao.java

```diff
@@ -151,60 +152,61 @@ default long insertComponentDef(long blockId, MailingAddressComponentDef mailAdd
     }
 
     default ComponentBlockDef findDefByBlockIdAndTimestamp(long blockId, long timestamp) {
-        ComponentDto componentDto = getJdbiComponent()
-                .findDtoByBlockIdAndTimestamp(blockId, timestamp)
-                .orElseThrow(() -> new DaoException(String.format(
-                        "Could not find component block definition for id %d and timestamp %d", blockId, timestamp)));
-
-        TemplateDao templateDao = getTemplateDao();
-
-        long componentId = componentDto.getComponentId();
-        if (componentDto.getComponentType() == ComponentType.MAILING_ADDRESS) {
-            MailingAddressComponentDto mailingAddressComponentDto = getJdbiComponent()
-                    .findMailingAddressComponentDtoById(componentId)
-                    .orElseThrow(() -> new DaoException("Could not find mailing address component with id " + componentId));
-            Template titleTmpl = null;
-            if (mailingAddressComponentDto.getTitleTemplateId() != null) {
-                titleTmpl = templateDao.loadTemplateById(mailingAddressComponentDto.getTitleTemplateId());
-            }
-            Template subtitleTmpl = null;
-            if (mailingAddressComponentDto.getSubtitleTemplateId() != null) {
-                subtitleTmpl = templateDao.loadTemplateById(mailingAddressComponentDto.getSubtitleTemplateId());
-            }
-            MailingAddressComponentDef comp = new MailingAddressComponentDef(titleTmpl, subtitleTmpl);
-            comp.setHideNumber(componentDto.shouldHideNumber());
-            comp.setRequireVerified(mailingAddressComponentDto.shouldRequireVerified());
-            comp.setRequirePhone(mailingAddressComponentDto.shouldRequirePhone());
-            return comp;
+        JdbiComponent jdbiComponent = getJdbiComponent();
+        Map<Long, Long> blockIdToComponentId = jdbiComponent
+                .findComponentIdsByBlockIdsAndTimestamp(Set.of(blockId), timestamp);
+        ComponentDto componentDto = jdbiComponent
```

Comment:
This is querying for a single component block right now. We'll bulk query for those in a later PR.